lib/Sema/ConstraintSystem.cpp

```diff
@@ -2490,8 +2490,16 @@ bool ConstraintSystem::diagnoseAmbiguityWithFixes(
                   diag::could_not_find_subscript_member_did_you_mean,
                   getType(UDE->getBase()));
     } else {
-      TC.diagnose(commonAnchor->getLoc(), diag::ambiguous_reference_to_decl,
-                  decl->getDescriptiveKind(), decl->getFullName());
+      auto name = decl->getBaseName();
```

Comment:
I'd love if this could be the full name in cases where there was only one full name, but I guess that's more work that it's worth right now. (Nice touch with the check for special names, though.)

Reply:
Actually it should be that bad, so I'm just going to do it in this PR.

Reply:
Done!